### PR TITLE
[16.0][ADD] sale_loyalty_exclude

### PR DESCRIPTION
**Pull request by santostelmo**
This module allows you to exclude products from a loyalty programs. These products have no effect on the minimum amount required to apply for the loyalty program.

### CODE REVIEW

**Review by cyrilmanuel — APPROVED** [review]

File: sale_loyalty_exclude/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _get_no_effect_on_threshold_lines(self):
+        """
+        Returns the lines that have no effect on the minimum amount to reach
+        """
+        self.ensure_one()
+        lines = self.order_line.filtered(lambda line: line.product_id.loyalty_exclude)
+        return lines + super()._get_no_effect_on_threshold_lines()
```
Comment (Ricardoalso) [inline sale_loyalty_exclude/models/sale_order.py]: Don't you want an `union` here instead of `+`?

**Review by Ricardoalso — APPROVED** [review]
Small comment, otherwise LG

File: sale_loyalty_exclude/models/sale_order.py (hunk 1 above)
Comment (santostelmo) [reply sale_loyalty_exclude/models/sale_order.py]: I think union and + are equivalent in the case because this is a recordset

Comment (Ricardoalso) [reply sale_loyalty_exclude/models/sale_order.py:16]: image
Apparently not

Comment (Ricardoalso) [reply sale_loyalty_exclude/models/sale_order.py]: image
image

Comment (santostelmo) [reply sale_loyalty_exclude/models/sale_order.py]: Interesting with `+` you have a duplicated record. I thought that was not possible 🤔

Comment (santostelmo) [reply sale_loyalty_exclude/models/sale_order.py]: Thank you @Ricardoalso in this case is better to use `|`

Comment (santostelmo) [reply sale_loyalty_exclude/models/sale_order.py]: In standard 17.0 they are using `+` https://github.com/odoo/odoo/blob/11bd9e01b79ec94c6e8e637a30cd412f01f2d791/addons/sale_loyalty_delivery/models/sale_order.py#L25

Comment (santostelmo) [reply sale_loyalty_exclude/models/sale_order.py]: Changed

**Pull request merged**